Testing/Unit/PowerShell/Orchestrator/Invoke-ScubaConfig.Tests.ps1

```diff
@@ -0,0 +1,100 @@
+$TestUtilsPath = "../PsTestUtils.psm1"
+Import-Module (Join-Path -Path $PSScriptRoot -ChildPath $OrchestratorPath) -Function 'Invoke-SCuBA' -Force
+Import-Module (Join-Path -Path $PSScriptRoot -ChildPath $ScubaConfigPath)
+Import-Module (Join-Path -Path $PSScriptRoot -ChildPath $TestUtilsPath)
+
+InModuleScope Orchestrator {
+    Describe -Tag 'Orchestrator' -Name 'Invoke-Scuba with Config' {
+        BeforeAll {
+            function Remove-Resources {}
+            Mock -ModuleName Orchestrator Remove-Resources {}
+            function Import-Resources {}
+            Mock -ModuleName Orchestrator Import-Resources {}
+            function Invoke-Connection {}
+            Mock -ModuleName Orchestrator Invoke-Connection { @() }
+            function Get-TenantDetail {}
+            Mock -ModuleName Orchestrator Get-TenantDetail { '{"DisplayName": "displayName"}' }
+            function Invoke-ProviderList {}
+            Mock -ModuleName Orchestrator Invoke-ProviderList {}
+            function Invoke-RunRego {}
+            Mock -ModuleName Orchestrator Invoke-RunRego {}
+
+            Mock -ModuleName Orchestrator Invoke-ReportCreation {}
+            function Disconnect-SCuBATenant {}
+            Mock -ModuleName Orchestrator Disconnect-SCuBATenant {}
+
+            Mock -CommandName New-Item {}
+            Mock -CommandName Copy-Item {}
+        }
+        Context 'Testing Invoke-Scuba with -ConfigFilePath arg and parameter override' {
+            BeforeAll {
+                $ConfigFile = ( Join-Path -Path $PSScriptRoot  -ChildPath "orchestrator_config_test.yaml" )
+                [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'SplatParamsRef')]
+                $SplatParamsRef = @{
+                    ConfigFilePath = $ConfigFile
+#                    M365Environment = 'gcc'
+                }
+                [ScubaConfig]::GetInstance().LoadConfig($ConfigFile)
+                $ScubaConfRef= [ScubaConfig]::GetInstance().Configuration.Clone()
+
+                function Override-Test( $ModKey, $ModValue) {
+                    $SplatParams = $SplatParamsRef.Clone()
+                    $SplatParams[$ModKey] = $ModValue
+                    Invoke-Scuba @SplatParams
+                    $ConfTest = [ScubaConfig]::GetInstance().Configuration
+                    # Verify there is a difference with the override
+                    $retTestForFalse = Compare-Hashes $ScubaConfRef $ConfTest | Select-Object -Last 1
+                    # Verify there are no other differences
+                    $ConfTest[$Modkey] = $ScubaConfRef[$Modkey]
+                    $retTestForTrue = Compare-Hashes $ScubaConfRef $ConfTest |  Select-Object -Last 1
+                    return ( -Not ( $retTestForFalse ) -and ( $retTestForTrue ))
+                }
+
+            }
+            It 'Verify override:  M365Environment -> gcc' {
```

Comment:
Compare Hash unit tests have been added with some enhancement in the function to support testing.  Note that the because clause was modified from the above test for this to function.